miner/worker.go

```diff
@@ -775,11 +785,234 @@ func (w *worker) updateSnapshot() {
 func (w *worker) commitTransaction(tx *types.Transaction, coinbase common.Address) ([]*types.Log, error) {
 	snap := w.current.state.Snapshot()
 
+	// reset StructLogger to avoid OOM
+	tracer := w.chain.GetVMConfig().Tracer.(*vm.StructLogger)
+	tracer.Reset()
+
+	zkTrieTracers := make(map[string]state.ZktrieProofTracer)
```

Comment:
The per account `zkTrieTracer` map should not be created on each tx commiting. It should be a field in work and being reset in `makeCurrent`. Also see the comment below.

Reply:
Note: it is not equivalent to merge all deletion proofs generated by the per-tx zktrieTracer since we use an algorithm which in fact scanned aproofs generated from all committed tx.